js/src/workspaces/window.js

```diff
@@ -395,7 +451,8 @@
 
       if (this.sidePanel === null) {
         this.sidePanel = new $.SidePanel({
-              parent: _this,
+              windowId: _this.id,
```

Comment:
There are a lot of examples of passing the window id and the global state from Window.  Widgets can be less dependent on Window, and simply publish events that trigger things in their Window.